### PR TITLE
Cleanup configuration shipping method after channel deletion

### PR DESCRIPTION
**Pull request by michalkaczmarek-bitbag**
| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

Ref:
```
After having deleted a channel in the admin view, I got form errors when editing the shipping methods.

I found out that the “configuration” column of the “sylius_shipping_method” table still contains the channel data as json, e.g.

{"WEB_EUR": {"amount": 960}, "FASHION_WEB": {"amount": 782}}

although both channels are deleted.

The users of our software need to delete the channel without errors.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Find shipping methods by channel-specific configuration.
  * Automatically remove channel configuration entries from shipping methods when a channel is deleted.
  * Added a channel-aware shipping-method updater and wired it into the channel deletion workflow.

* **Tests**
  * Added tests for shipping-method lookup by channel code.
  * Added tests for channel-configuration removal (including flush behavior and edge cases).
  * Added fixtures for shipping methods and zones used by tests.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php [hunk 1]
```diff
@@ -13,15 +13,20 @@
...
 use Sylius\Resource\Symfony\EventDispatcher\GenericEvent;
 
 final class ChannelDeletionListener
 {
-    public function __construct(private ChannelDeletionCheckerInterface $channelDeletionChecker)
-    {
+    public function __construct(
+        private ChannelDeletionCheckerInterface $channelDeletionChecker,
+        private ShippingMethodRepositoryInterface $shippingMethodRepository,
+        private EntityManagerInterface $entityManager,
```
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php]: Introduce a new, dedicated service (and an Autowireable alias, check other examples in the codebase):

I'm proposing the following namespaces:
```php
// Interface
  Sylius\Component\Core\ShippingMethod\Updater\ChannelAwareShippingMethodUpdaterInterface

// Implementation
  Sylius\Bundle\CoreBundle\ShippingMethod\Updater\ShippingMethodUpdater
```

And the configuration:
```
<service id="sylius.updater.shipping_method"
  class="Sylius\Bundle\CoreBundle\ShippingMethod\Updater\ShippingMethodUpdater">
      <argument type="service" id="sylius.repository.shipping_method" />
      <argument type="service" id="doctrine.orm.entity_manager" />
  </service>
  <service id="Sylius\Component\Core\ShippingMethod\Updater\ChannelAwareShippingMethodUpdaterInterface"
  alias="sylius.updater.shipping_method" />
```

(Create a new directory)
```php
namespace Sylius\Component\Core\ShippingMethod\Updater;
```

```php
  final class ShippingMethodUpdater implements ChannelAwareShippingMethodUpdaterInterface
  {
      // ...
  }
```

 and utilize it here:

```php
use Sylius\Component\Core\Updater\ChannelAwareShippingMethodUpdaterInterface;
```

```suggestion
    public function __construct(
        private ChannelDeletionCheckerInterface $channelDeletionChecker,
        private ChannelAwareShippingMethodUpdaterInterface $shippingMethodUpdater,
```


Remember to adjust the tests after that - both the simple listener test and the full coverage for the new service.

File: src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php [hunk 2]
```diff
@@ -42,4 +47,31 @@ public function onChannelPreDelete(GenericEvent $event): void
...
+    public function removeChannelConfigurationFromShippingMethods(GenericEvent $event): void
+    {
+        $channel = $event->getSubject();
+
+        if (!$channel instanceof ChannelInterface) {
+            throw new UnexpectedTypeException(
+                $channel,
+                ChannelInterface::class,
+            );
+        }
```
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php]: Could be extracted to a common private method.

**Review by Rafikooo — APPROVED** [review]
